src/composables/useDefaultPreset.ts

```diff
@@ -0,0 +1,27 @@
+import { voiceToVoiceId } from "@/lib/voice";
+import { useStore } from "@/store";
+import { Voice } from "@/type/preload";
+import { computed } from "vue";
+
+export const useDefaultPreset = () => {
```

Comment:
`src/composables` を導入しました。
各コンポーネント間で共通だけどVuex Storeに乗せる処理ではないな～でもコピペするのもな～という単位の処理を切り出すのによさそうだったので。

https://ja.vuejs.org/guide/reusability/composables.html

Reply:
結局同じ処理をvuex側に書いてたりするのでこれはpreset.tsに書いた方がよかったかもですね・・・
MUSTでなければリファクタPRで良い感じにします、という感じの落としどころにさせてください

Reply:
なるほどです！　とりあえず他の方が見たときにわかりやすい＆githubで足すのが容易（apply suggestionするだけ）なので、FIXMEだけ書き足すのはどうでしょう。
```suggestion
// FIXME: vuex側から使いたいこともあるので、`store/preset.ts`に移す
export const useDefaultPreset = () => {
```

Reply:
すみませんコード見直してみたのですが、これはこのcomposable内でcacheとして一時的にSetを持っておき、何度もisDefaultPresetで判定するときのコストを減らす意図だったようです。
（`presets.filter(isDefaultPreset)` という判定してたと思うので毎回リストなめてたらO(n^2)になってしまう）

たぶんisDefaultPresetが他のところにも現れているから共通化、という話になってると思うので、目的が違うので共通化はしない、という答えになります。
vuex側は1カ所しか使われていないので、複数の箇所で使われるようになったら切り出せばよいと思います

Reply:
なるほどです！！　たしかにスタイル数×プリセット数の計算が走って、それは重い気がしました。

ちょっと確かめてみたのですが、もしかしたらいまのコードだと`composable`内でキャッシュ化できてないかもです。
`isDefaultPresetKey`内でconsole.logしてみた感じ、同じ`presetKey`で何度も関数が呼ばれてそうでした。

> 読み込み時に計算して別途defaultPresetKeySetsを

なるほどです！
[Vuexのgetterはcomputed的な機能がある](https://vuex.vuejs.org/ja/guide/getters.html)ので、getterに載せちゃうのがスマートかもと思いました！！

Reply:
了解です、リファクタPRで修正します

Reply:
と書いておきながら、もしそうだったらさすがに無意味な記述なのでこのPR内で直さんとだめかな・・・と調査しました。
キャッシュという言葉を出してしまったのが悪かった気がするのですが、たぶん認識が違ってそうです。

`isDefaultPresetKey` はただの関数呼び出しなので、それはもちろん何度も呼ばれます。
`isDefaultPresetKey` 内で毎回O(n^2)をやらなくて良いように予め計算したSet（これがキャッシュです）を持つことで
`isDefaultPresetKey` の計算コストをO(1)にする、というのがこの関数の目的です。

もし `new Set(Object.values(store.state.defaultPresetKeys));` の部分が毎回計算されていたら「キャッシュされていない」と言えますが、そうはなっていないことを確認しました。
ということでキャッシュ化できています。
このキャッシュはグローバルに置いている訳ではないので、キャッシュのの寿命＝コンポーネントの寿命になっていますが、これも意図通りです。

Reply:
あ、なるほどです！！
Set化するとこに意味があることが頭から抜けていた＋`has`の計算量がO(N)と意図を間違えていました、なるほどです！！